### ISSUE TITLE
Capability Delegation for getDisplayMedia()

### ISSUE THREAD

**Issue by beaufortfrancois** [WebKit/standards-positions#137]
## Request for position on an emerging web specification

* WebKittens who can provide input: @youennf 

### Information about the specification

* Title: Capability Delegation for getDisplayMedia()
* URL: https://wicg.github.io/capability-delegation/spec.html#monkey-patch-to-screen-capture
* GitHub repository: https://wicg.github.io/capability-delegation/spec.html
* Explainer: https://github.com/WICG/capability-delegation#allowing-display-capture-from-cross-origin-iframe-click

### Design reviews and vendor positions <ins>for the general delegation mechanism</ins>

* TAG Design Review: https://github.com/w3ctag/design-reviews/issues/655
* Mozilla standards-positions issue: https://github.com/mozilla/standards-positions/issues/565

### Anything else we need to know

TLDR; Capability Delegation for getDisplayMedia() allows third-party `iframe` to prompt the user to share a tab, a window, or a screen when user interacts with the top frame by delegating the `mediaDevices.getDisplayMedia()` permission to the iframe as follows:

```js
// In the top frame, user clicks the "Share My Screen" button.
button.onclick = () =>
  frames[0].postMessage("msg", { delegate: "display-capture" });
```
```js
// In the cross-origin video-conferencing iframe, prompt the user
// to share a tab, a window, or a screen.
window.onmessage = () => navigator.mediaDevices.getDisplayMedia();
```

cc @eladalon1983 @mustaqahmed

**Comment by annevk** [comment]
My primary concern is that https://github.com/WICG/capability-delegation/issues has had substantive issues open for well over a year.

I don't think it's responsible for Chromium to continue to build on top of it when it's clearly not being maintained to the extent that it needs to be.

**Comment by beaufortfrancois** [comment]
The Chrome team is currently looking into prioritising this work. See https://github.com/mozilla/standards-positions/issues/565#issuecomment-1442142491

**Comment by hober** [comment]
> We should probably first come up on a position about capability-delegations. I do not see an issue filed for capability delegation, would it help to do this?

Please do!

**Comment by beaufortfrancois** [comment]
@mustaqahmed filed one at https://lists.webkit.org/pipermail/webkit-dev/2022-May/032221.html

**Comment by youennf** [comment]
@beaufortfrancois or @mustaqahmed, would you be willing to file a new issue for that purpose, with the latest available information.
Otherwise, I'll do it based on this email sometimes before the end of this week.

**Comment by mustaqahmed** [comment]
I will take care of a new issue for core Capability Delegation support, ETA sometime next week.  Thanks @beaufortfrancois and @youennf for bringing this to my attention.
